collector/perf_schema_memory_events.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2020 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Scrape `performance_schema.memory_summary_global_by_event_name`.
+
+package collector
+
+import (
+	"context"
+	"database/sql"
+	"strings"
+
+	"github.com/go-kit/kit/log"
+	"github.com/prometheus/client_golang/prometheus"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+const perfMemoryEventsQuery = `
+	SELECT
+		EVENT_NAME, SUM_NUMBER_OF_BYTES_ALLOC, SUM_NUMBER_OF_BYTES_FREE,
+		CURRENT_NUMBER_OF_BYTES_USED
+	FROM performance_schema.memory_summary_global_by_event_name
+		where COUNT_ALLOC > 0;
+`
+
+// Tunable flags.
+var (
+	performanceSchemaMemoryEventsRemovePrefix = kingpin.Flag(
+		"collect.perf_schema.memory_events.remove_prefix",
+		"Remove instrument prefix in performance_schema.memory_summary_global_by_event_name",
+	).Default("memory/").String()
+)
+
+// Metric descriptors.
+var (
+	performanceSchemaMemoryBytesAllocDesc = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, performanceSchema, "memory_events_bytes_alloc_total"),
```

Comment:
We like to put the unit at the end.

```suggestion
		prometheus.BuildFQName(namespace, performanceSchema, "memory_events_alloc_bytes_total"),
```